### PR TITLE
Speed up PO box test

### PR DESCRIPTION
**Pull request by zmbc**
### Description

- *Category*: performance, test
- *JIRA issue*: none
- *Research reference*: none

### Changes and notes

Probably tons of other low-hanging fruit but I noticed this one because it was new.

### Verification and Testing

This test went from taking over 1.5 minutes (not including simulation running time, just the test body itself) to a few seconds.

### CODE REVIEW

File: integration_tests/test_domestic_migration.py [hunk 1]
```diff
@@ -268,17 +268,15 @@ def test_addresses_during_moves(simulants_on_adjacent_timesteps, unit_id_col, ad
...
         assert np.isclose(
-            len(
-                pop[pop["household_details.po_box"] == data_values.NO_PO_BOX].groupby(
-                    "household_id"
-                )
-            )
-            / len(pop.groupby("household_id")),
+            pop[
+                pop["household_details.po_box"] == data_values.NO_PO_BOX
+            ].household_id.nunique()
```
Comment (rmudambi) [inline integration_tests/test_domestic_migration.py:278]: Use bracket notation rather than dot notation in both of these lines

**Review by rmudambi — APPROVED** [review]
Thanks for catching this!

File: integration_tests/test_domestic_migration.py (hunk 1 above)
Comment (mattkappel) [reply integration_tests/test_domestic_migration.py:278]: +1

**Comment by mattkappel** [comment]
Thanks much. TIL.

**Review by mattkappel — APPROVED** [review]

**Comment by mattkappel** [comment]
It still seems weird to me that `nunique` is more efficient than `len` but perhaps that's because I'm doing it on the entire data frame rather than a series.

**Comment by aflaxman** [comment]
> It still seems weird to me that `nunique` is more efficient than `len` but perhaps that's because I'm doing it on the entire data frame rather than a series.

I think the speed up comes from avoiding the `.groupby`

File: integration_tests/test_domestic_migration.py (hunk 1 above)
Comment (zmbc) [reply integration_tests/test_domestic_migration.py:278]: Ah duh, thanks for the reminder!

**Pull request merged**